Automattic/jetpack
# Classic Theme Helper Plugin: Ensure Featured Content and imported / required files in the package are initialized

Fixes https://github.com/Automattic/vulcan/issues/384

## Proposed changes:

* This PR ensures the classes that initialize both Featured Content and Responsive Videos from the Classic Theme Helper package are initialized in the plugin, so they can work even if Jetpack is not also initializing those classes.

This PR only considers the features that have been added to the Classic Theme Helper package and focuses on loading them when the plugin is active. More may be needed on the plugin generally, but that can be covered in later work: https://github.com/Automattic/vulcan/issues/388

This also assumes the current version of Jetpack in use on a site is the equivalent of trunk, as it isn't expected this plugin would be publicly used (with the current latest stable v of Jetpack or 13.7) until around Jetpack 13.8 or later. We may want to gate requiring / initializing files and classes to certain Jetpack versions to avoid issues, but that may need some discussion - mentioned here: https://github.com/Automattic/vulcan/issues/388

### Other information:


## Jetpack product discussion

https://github.com/Automattic/vulcan/issues/384

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

Replicating the issue:

* If testing locally, make sure you have installed and built the Classic Helper Theme plugin:  `jetpack install plugins/classic-theme-helper-plugin`, `jetpack build plugins/classic-theme-helper-plugin`, then activate the plugin.
* Also if testing locally, remove `"automattic/jetpack-classic-theme-helper": "@dev",` from the Jetpack plugin `composer.json` file. Then run run `tools/fixup-project-versions.sh`, and then `jetpack install plugins/jetpack` and `jetpack build plugins/jetpack`.
* If testing using the Beta tester plugin, activate 'Bleeding Edge' for the Classic Theme Helper plugin.
* Comment out `theme-tools/responsive-videos.php'` and `theme-tools/featured-content.php` from `projects/plugins/jetpack/modules/module-extras.php` (via SFTP if using a test site and the Beta tester plugin)
* If testing using the Beta tester plugin,  delete the `jetpack-classic-theme-helper` package folder from `jetpack/jetpack-vendor/automattic`.
* Follow the steps in this PR to test Responsive Videos and Featured Content (only on self-hosted): https://github.com/Automattic/jetpack/pull/37969
* They shouldn't work.

To test the fix:

* Apply this PR if testing locally or using the Beta tester plugin (activating the branch for the Classic Theme Helper plugin).
* Follow the same steps as above (from 'comment out' if using the Beta tester plugin)
* Responsive Videos and Featured Content should work as expected - and the source of the scripts when viewed in page source should be the Classic Theme Helper plugin (not the Jetpack Mu Wpcom plugin or Jetpack).
